ArduSub/control_poshold.cpp

```diff
@@ -63,19 +62,24 @@ void Sub::poshold_run()
     float lateral_out = 0;
     float forward_out = 0;
 
-    // Allow pilot to reposition the sub
-    if (fabsf(pilot_lateral) > 0.1 || fabsf(pilot_forward) > 0.1) {
-        lateral_out = pilot_lateral;
-        forward_out = pilot_forward;
-        loiter_nav.clear_pilot_desired_acceleration();
-        loiter_nav.init_target(); // initialize target to current position after repositioning
+    pos_control.set_desired_velocity_xy(0,0);
+
+    // TODO: check if 'visual_odom.healthy' is required for real hardware
+    if (visual_odom.healthy() || position_ok()) {
```

Comment:
i think this should be only position_ok(). if necessary, we can add the visual_odom healthy check to Sub::optflow_position_ok logic

Reply:
It was just my memory failing on me, it works with position_ok() now:
https://github.com/ArduPilot/ardupilot/pull/13982#issuecomment-609921878